commit 8350ff933c37faa15101662516d8614a9c301a36
Author: Pierre Neidhardt <mail@ambrevar.xyz>
Date:   Thu Mar 5 16:43:55 2020 +0100

    Work around Guix compilation error of define-key.
    
    Error happens in search-buffer's search-over-buffers which calls define-key.
    
    ; in: DEFUN SEARCH-OVER-BUFFERS => DEFINE-KEY "C-s"
    ;     (NEXT:DEFINE-KEY "C-s"
    ;      #'(LAMBDA () (NEXT::UPDATE-SELECTION-HIGHLIGHT-HINT :FOLLOW T :SCROLL T))
    ;      :KEYMAP NEXT::KEYMAP)
    ;
    ; caught WARNING:
    ;   The first argument of type (SIMPLE-ARRAY CHARACTER (3)) cannot be used as a keyword.
    
    Maybe the type is wrong.  But our define-key lambdalist is weird anyways.

diff --git a/source/keymap.lisp b/source/keymap.lisp
index bbcb9d21..e29bdf3a 100644
--- source/keymap.lisp
+++ source/keymap.lisp
@@ -144,7 +144,7 @@
          (setf (key-chord-stack *browser*) nil))
         (t (setf (key-chord-stack *browser*) nil))))))
 
-(declaim (ftype (function (&rest t &key (:scheme list) (:keymap keymap) &allow-other-keys)) define-key))
+;; (declaim (ftype (function (&rest t &key (:scheme list) (:keymap keymap) &allow-other-keys)) define-key)) ; TODO: This fails with Guix.
 @export
 (defun define-key (&rest key-command-pairs
                    &key keymap
